src/dx/pubnub_client.rs

```diff
@@ -131,8 +131,11 @@ where
     pub(crate) transport: T,
 
     /// Instance ID
-    #[builder(setter(strip_option), default = "None")]
-    pub(crate) instance_id: Option<String>,
+    #[builder(field(
+        type = "&'static str",
+        build = "Arc::new(Some(self.instance_id.to_string()))"
+    ))]
```

Comment:
Now client cannot use Strings for setting this field. 
You can allow that by:
```suggestion
#[builder(
        setter(into),
        field(type = "String", build = "Arc::new(Some(self.instance_id))")
    )]
```